lib/action_client/callbacks/single_argument_callback.rb

```diff
@@ -0,0 +1,23 @@
+module ActionClient
+  module Callbacks
+    SingleArgumentCallback = Struct.new(:status, :headers, :body) do
+      def call(block)
+        modified_body = block.call(body).tap do |response|
+          response = Array(response)
+
+          if response.length != 1
```

Comment:
@georgebrock the more I think about these guards, the less confident I am in their value.

Consider the following:

```ruby
class ArticlesClient < ActionClient::Base
  def all
    get url: "https://example.com/articles" do |body|
      body.map { |attributes| Article.new(attributes) }
    end
  end
end

stub_request(:get, %r{example.com}).to_return(body: [{ id: 1 }].to_json)

status, headers, body = ArticlesClient.all.submit
```

In this scenario, the callback returns _only_ the body, but the body happens to be a single item array.

Since controlling the data set returned by an external server is outside of any application codebase's control, this could be a surprising and dangerous source of errors in production environments.

I'm not sure if the same scenario is possible for the triplet style callbacks, so maybe we can keep that guard in place.

What do you think?